src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.cs

```diff
@@ -760,11 +757,7 @@ public virtual bool IsSuffix(string source, string suffix, CompareOptions option
 
             if (suffix.Length == 0)
             {
-                return true;
-            }
-            if (source.Length == 0)
```

Comment:
same we should keep this condition for ordinal cases.

Reply:
Fixed in most recent iteration.